.github/workflows/gradle_publish.yml

```diff
@@ -0,0 +1,96 @@
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#Publishing-using-gradle
+
+name: Gradle Package
+
+on:
+  release:
+    types: [ created ]
+  workflow_dispatch:
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get mod version from Gradle
+        uses: madhead/read-java-properties@latest
+        id: forgero_version
+        with:
+          file: gradle.properties
+          property: mod_version
+          default: 0.0.1
+
+      - name: Get supported Minecraft version from Gradle
+        uses: madhead/read-java-properties@latest
+        id: supported_minecraft_version
+        with:
+          file: gradle.properties
+          property: minecraft_version
+          default: 0.0.1
+
+      - run: echo version string ${{ steps.mod_version.outputs.value }}+mc${{ steps.supported_minecraft_version.outputs.value }}
+
+      - name: Set up Git
+        run: |
+          git --version
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+          git status
+          git describe --tags
+        shell: bash
+
+      - name: Create Git release tag
+        run: |
+          git tag ${{ steps.mod_version.outputs.value }}+mc${{ steps.supported_minecraft_version.outputs.value }}
+          git push origin ${{ steps.mod_version.outputs.value }}+mc${{ steps.supported_minecraft_version.outputs.value }}
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
+          settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Execute Gradle build
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build
+
+      - name: Upload Gradle artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: compiled_jar_files
+          path: fabric/build/libs/
+          if-no-files-found: error
+
+      # The USERNAME and TOKEN need to correspond to the credentials environment variables used in
+      # the publishing section of your build.gradle
+      - name: Publish to GitHub Packages
+        uses: gradle/gradle-build-action@937999e9cc2425eddc7fd62d1053baf041147db7
+        with:
+          arguments: publish
+        env:
+          GITHUB_ACTOR: ${{ github.actor }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish to CurseForge
+        run: ./gradlew curseforge
+        env:
+          CurseForgeApiKey: ${{ secrets.CURSEFORGEAPIKEY }}
```

Comment:
I suggest using CI with a manual workflow run for this, since we can still test locally, and then build with proper tags and filenames in CI.

Reply:
If we use the task `publishToModrinthAndCurseForge` it will upload to both curse and modrinth.

Env's would be `MODRINTH_TOKEN` and `CURSEFORGE_API_KEY`